hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/TestSparkRDDMetadataWriteClient.java

```diff
@@ -0,0 +1,414 @@
+    initDataTableWithACommit(hoodieWriteConfig);
+
+    // fetch metadata file slice info
+    HoodieWriteConfig mdtWriteConfig = HoodieMetadataWriteUtils.createMetadataWriteConfig(hoodieWriteConfig, HoodieFailedWritesCleaningPolicy.EAGER);
+    Map<String, List<String>> mdtPartitionsFileIdMapping = new HashMap<>();
+    List<HoodieFileGroupId> nonFilesPartitionFileGroupIdList = new ArrayList<>();
+    List<HoodieFileGroupId> filesPartitionFileGroupIdList = new ArrayList<>();
+    HoodieTableMetaClient metadataMetaClient = HoodieTableMetaClient.builder().setBasePath(mdtWriteConfig.getBasePath()).setConf(context.getStorageConf()).build();
+    fetchMetadataFileSliceInfo(metadataMetaClient, filesPartitionFileGroupIdList, nonFilesPartitionFileGroupIdList, mdtPartitionsFileIdMapping);
+
+    List<HoodieRecord> filesPartitionExpectedRecords = new ArrayList<>();
+    Map<String, HoodieRecord> filesPartitionExpectedRecordsMap = new HashMap<>();
+    List<String> expectedAllPartitions = new ArrayList<>();
+    List<HoodieRecord> rliRecords = new ArrayList<>();
+    Map<String, HoodieRecord> rliPartitionExpectedRecordsMap = new HashMap<>();
+    String commitTimeOfInterest = null;
+
+    // create Write client to SparkRDDMetadataWriteClient and trigger writes.
+    try (SparkRDDMetadataWriteClient client = new SparkRDDMetadataWriteClient(context, mdtWriteConfig)) {
+      commitTimeOfInterest = client.createNewInstantTime();
+
+      // prepare FILES partition records.
+      prepareFilesPartitionRecords(mdtPartitionsFileIdMapping.get(MetadataPartitionType.FILES.getPartitionPath()).get(0),
+          commitTimeOfInterest, filesPartitionExpectedRecords, filesPartitionExpectedRecordsMap, expectedAllPartitions);
+
+      // prepare RLI records.
+      prepareRliRecords(commitTimeOfInterest, mdtPartitionsFileIdMapping, expectedAllPartitions, rliRecords, rliPartitionExpectedRecordsMap);
+
+      // ingest RLI records to metadata table.
+      client.startCommitForMetadataTable(metadataMetaClient, commitTimeOfInterest, DELTA_COMMIT_ACTION);
+      JavaRDD<WriteStatus> partialWriteStatusesRDD = client.upsertPreppedRecords(jsc.parallelize(rliRecords), commitTimeOfInterest, Option.of(nonFilesPartitionFileGroupIdList));
+      List<WriteStatus> partialWriteStatuses = partialWriteStatusesRDD.collect();
+
+      // validate that the commit is still pending since we are streaming write to metadata table.
+      HoodieActiveTimeline reloadedMdtActiveTimeline = metadataMetaClient.reloadActiveTimeline();
+      assertEquals(reloadedMdtActiveTimeline.filterCompletedInstants().getInstants().stream().count(), 3); // files, rli instantiaton and 1 write to data table.
+      String finalCommitTimeOfInterest = commitTimeOfInterest;
+      assertTrue(reloadedMdtActiveTimeline.filterInflightsAndRequested().getInstants().stream().anyMatch(instant -> instant.requestedTime().equals(finalCommitTimeOfInterest)));
```

Comment:
If there are two `.inflight` files for a commit, will that throw an exception somewhere? Wondering if we want to validate this by listing the directory manually and inspecting it.